### PR TITLE
Added support for enhanced animations for items.

### PR DESCRIPTION
**Pull request by Mignari**
Thanks to @conde2, @BenDol

### CODE REVIEW

**Comment by TheSumm** [comment]
Good job, thing.h still has: 
    Animation getAnimation() { return rawGetThingType()->getAnimation(); }

**Comment by totolol123** [comment]
Awesome :+1:

**Comment by BenDol** [comment]
Thanks a lot @Mignari

File: src/client/thingtype.cpp [hunk 1]
```diff
@@ -303,19 +297,48 @@ void ThingType::unserialize(uint16 clientId, ThingCategory category, const FileS
...
+
+                for(int i = 0; i < m_animationPhases; i++) {
+                    int minimum = fin->getU32();
+                    int maximum = fin->getU32();
+                    phaseDurations.push_back(std::make_tuple(minimum, maximum));
+                }
+            }
+            else {
+                int duration = 0;
+                switch (m_category) {
```
Comment (BenDol) [inline src/client/thingtype.cpp]: remove the space

File: src/client/animator.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    m_currentDirection = AnimDirForward;
+    m_currentLoop = 0;
+
+    m_lastPhaseTicks = 0;
+    m_isComplete = false;
+
+    m_phase = getStartPhase();
+}
+
+void Animator::serialize(const FileStreamPtr& fin)
```
Comment (BenDol) [inline src/client/animator.cpp]: Would it be easy enough to add unserialize method? Can be added after the PR anyway though.

**Comment by Nu77** [comment]
Nice job :+1:

**Comment by Mignari** [comment]
Please don't merge it. Just let me know if needs more changes. I want to create a single commit to this PR.

**Comment by BenDol** [comment]
LGTM

**Comment by Mignari** [comment]
Closing to create new pull request.

**Comment by BenDol** [comment]
Why create a new PR? The whole point is for feedback and updating then merging :P

**Comment by Mignari** [comment]
I prefer it that way. Because I see no reason to merge 4 horrible commits to a single resource.

**Comment by BenDol** [comment]
Okay LGTM

**Pull request closed without merging**